### ISSUE TITLE
Feature Request: Case folding

### ISSUE THREAD

**Issue by Quackdoc** [svenstaro/miniserve#1445]
It would be nice if miniserve supported casefolding itself, if you try do deploy something that was developed on windows, filenames will often break due to case sensitivity

**Comment by svenstaro** [comment]
Let's be sure that I know what you mean: You request a file `hello.omg` at `http://localhost:8000/heLLo.omg` and with case-folding turned on, it would still serve it?

**Comment by Quackdoc** [comment]
Correct. This is fairly common if you have a server on a Windows host and try to migrate it to a Linux host. There are some file systems on linux which do support case folding, which does fix the issue, but not all file systems supported.

**Comment by svenstaro** [comment]
What would the interaction be on a case-sensitive filesystem with this option turned on? If there's a dir with `Hello.omg` and `hellO.omg`, which one would be used? Do you know of any webserver that does this so we can look at some prior art?

**Comment by Quackdoc** [comment]
I'm not sure if there are webservers that do this, and which one it chooses may entirely be up to miniserve to figure out (possibly even with a flag that swaps between lower first or upper first?) as for "prior art" the closest example I have is this python script which starts the http.server with case insensitivity.

[case.txt](https://github.com/user-attachments/files/16362405/case.txt)
